azurelinuxagent/common/utils/networkutil.py

```diff
@@ -155,6 +185,18 @@ def __get_common_drop_command_params(wait, command, destination):
         cmd.extend(["conntrack", "--ctstate", "INVALID,NEW", "-j", "DROP"])
         return AddFirewallRules._add_wait(wait, cmd)
 
+    # This rule allows DNS TCP request to wireserver ip for non root users
+    @staticmethod
+    def get_iptables_accept_dns_tcp_request_command(wait, command, destination):
+        cmd = AddFirewallRules.__get_iptables_base_command(wait)
+        if command == AddFirewallRules.get_insert_command():
+            cmd.extend(['-t', 'security', '-I', 'OUTPUT', '-d', destination, '-p', 'tcp', '--destination-port', '53', '-j', 'ACCEPT'])
+        if command == AddFirewallRules.get_check_command():
```

Comment:
Please use + instead of extend()... right now this works because __get_iptables_base_command always returns a new instance of the array, but if that changes this code would keep extending the same array.

Reply:
Sure, Addressed this in the next commit